controllers/storagecluster/cephfilesystem.go

```diff
@@ -59,6 +42,33 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initData *ocsv1.St
 			},
 		},
 	}
+
+	dataPools := make([]cephv1.NamedPoolSpec, len(initData.Spec.StorageDeviceSets))
+	if len(initData.Spec.StorageDeviceSets) > 0 {
+		for i := range initData.Spec.StorageDeviceSets {
+			deviceClass := initData.Spec.StorageDeviceSets[i].DeviceClass
+			var parameters map[string]string
+			// choose the right ceph parameters for the deviceclass
+			for i := range initData.Spec.StorageProfiles {
+				if initData.Spec.StorageProfiles[i].DeviceClass == deviceClass {
+					parameters = initData.Spec.StorageProfiles[i].SharedFilesystemConfiguration.Parameters
+					break
+				}
+			}
+			dataPools[i] = cephv1.NamedPoolSpec{
+				Name: deviceClass,
+				PoolSpec: cephv1.PoolSpec{
+					Replicated:    generateCephReplicatedSpec(initData, "data"),
+					DeviceClass:   deviceClass,
+					Parameters:    parameters,
+					FailureDomain: initData.Status.FailureDomain,
+				},
+			}
+		}
+	}
```

Comment:
Not using append as we need to support a standalone (upgrades) cluster as it will not have a storage profile but needs a default data pool. If the storage profile is defined, we will replace it.